src/worldstate/routine/src/State.cpp

```diff
@@ -0,0 +1,32 @@
+//
+// Created by joel on 12/05/18.
+//
+
+#include "State.h"
+
+State::State(int argc, char** argv, std::string node_name) {
+    ros::init(argc, argv, node_name);
+}
+
+void State::start() {
+    ros::NodeHandle nh;
+    setupNodeSubscriptions(nh);
+
+    ros::NodeHandle private_nh("~");
+
+    std::string state_transition_msg =
+    private_nh.resolveName("worldstate/output");
+    int queue_size_  = 1;
+    state_publisher_ = private_nh.advertise<worldstate::state_msg>(
+    state_transition_msg, queue_size_);
+
+    // ros::spin();
+}
+
+void State::sleep() {
+    ros::shutdown();
```

Comment:
I think that running `ros::shutDown` will kill the node that spawned this one as well (ie. it would kill the manager as well). Killing each state should probably be the responsbility of the manager (ie. the StateManager should see a state transition, stop the current state node, and start the next one)

Reply:
This was probably copied over from my design where I call this as well, from my research it seemed like the way to go but I will do some testing this weekend to see how this works.